From 06f86bae9a0b94850d6006fb7b65b8ecf8d57dcd Mon Sep 17 00:00:00 2001
From: Kaleb S KEITHLEY <kkeithle@redhat.com>
Date: Sun, 15 May 2016 11:06:08 -0400
Subject: [PATCH 153/158] common-ha: floating IP (VIP) doesn't fail over when ganesha.nfsd dies

restore mistaken removal of 'attrd_updater delete grace-active' to
trigger fail-over

original was:
   attrd_updater -D -n grace-active
   sleep
   attrd_updater -D -n ganesha-active

mistake was:
   sleep
   attrd_updater -D -n grace-active

Change-Id: I0c69c363301735d3ff150e13637e715240a343fc
BUG: 1335826
Signed-off-by: Kaleb S KEITHLEY <kkeithle@redhat.com>
Reviewed-on: https://code.engineering.redhat.com/gerrit/74330
Reviewed-by: Soumya Koduri <skoduri@redhat.com>
Tested-by: Soumya Koduri <skoduri@redhat.com>
---
 extras/ganesha/ocf/ganesha_mon |    9 +++++++--
 1 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/extras/ganesha/ocf/ganesha_mon b/extras/ganesha/ocf/ganesha_mon
index 65450b8..dbb9eaa 100644
--- a/extras/ganesha/ocf/ganesha_mon
+++ b/extras/ganesha/ocf/ganesha_mon
@@ -173,8 +173,6 @@ ganesha_mon_monitor()
 	# the remaining ganesha.nfsds into grace before
 	# initiating the VIP fail-over.
 
-	sleep ${OCF_RESKEY_grace_delay}
-
 	attrd_updater -D -n ${OCF_RESKEY_grace_active}
 	if [ $? -ne 0 ]; then
 		ocf_log info "warning: attrd_updater -D -n ${OCF_RESKEY_grace_active} failed"
@@ -185,6 +183,13 @@ ganesha_mon_monitor()
 		ocf_log info "warning: crm_attribute --node=${short_host} --name=${OCF_RESKEY_grace_active} --update=0 failed"
 	fi
 
+	sleep ${OCF_RESKEY_grace_delay}
+
+	attrd_updater -D -n ${OCF_RESKEY_ganesha_active}
+	if [ $? -ne 0 ]; then
+		ocf_log info "warning: attrd_updater -D -n ${OCF_RESKEY_ganesha_active} failed"
+	fi
+
 	return ${OCF_SUCCESS}
 }
 
-- 
1.7.1

